src/main/java/com/deavensoft/paketomat/center/CenterServiceImpl.java

```diff
@@ -1,46 +1,54 @@
 package com.deavensoft.paketomat.center;
 
 import com.deavensoft.paketomat.center.model.Package;
+import com.deavensoft.paketomat.center.model.Status;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.stereotype.Service;
 
+import java.util.ArrayList;
 import java.util.List;
 import java.util.Optional;
 
 @Service
-public class CenterServiceImpl implements CenterService{
+public class CenterServiceImpl implements CenterService {
     private CenterRepository centerrepository;
     private com.deavensoft.paketomat.center.model.Center distributiveCenter;
+
+
+
     @Autowired
-    public CenterServiceImpl(@Qualifier ("center") CenterRepository centerrepository)
-    {
+    public CenterServiceImpl(@Qualifier("center") CenterRepository centerrepository) {
         this.centerrepository = centerrepository;
         this.distributiveCenter = new com.deavensoft.paketomat.center.model.Center();
         initialise();
     }
-    public void initialise(){
+
+    public void initialise() {
 
     }
-    public List<Package> getAllPackages()
-    {
-      return  centerrepository.findAll();
+
+    public List<Package> getAllPackages() {
+        return centerrepository.findAll();
     }
 
-    public void save(Package packagee)
-    {
+    public void save(Package packagee) {
         centerrepository.save(packagee);
     }
 
-    public Optional<Package> findPackageById(Long id)
-    {
+    public Optional<Package> findPackageById(Long id) {
         return centerrepository.findById(id);
     }
-    public void deletePackageById(Long id)
-    {
+
+    public List<Package> findAllPackageByStatus(Status status){
```

Comment:
Why do we need the recursive call of this method? Do we need this method at all?